lua/autorun/server/sv_chip_lister.lua

```diff
@@ -0,0 +1,240 @@
+
+        return context and context.timebench or 0
+    end
+
+    return 0
+end
+
+local function normalizeCPUs( cpus )
+    return mRound( ( cpus or 0 ) * CPU_MULT )
+end
+
+local function prepareName( str, maxLength )
+    str = stringTrim( stringReplace( str or "", "\n", " " ) )
+    
+    if stringLen( str ) > maxLength then
+        str = stringSub( str, 1, maxLength - 3 ) .. "..."
+    end
+
+    return str
+end
+
+local function preparePlyName( str )
+    return prepareName( str, PLAYER_LENGTH_MAX )
+end
+
+local function prepareChipName( str )
+    return prepareName( str, CHIP_LENGTH_MAX )
+end
+
+local function updateChipLister()
+    if listUserCount == 0 then return end
+
+    local perPlyData = {}
+    local idLookup = {}
+    local globalUsage = 0
+    local idCount = 0
+
+    for i = 1, chipCount do
+        local chip = rawget( chips, i )
+
+        if isValid( chip ) then
+            local chipName = prepareChipName( " -" .. getChipName( chip ) )
+            local chipClass = getClass( chip )
+            local chipUsage = normalizeCPUs( getCPUs( chip, chipClass ) )
+            local owner = getOwner( chip )
+            local ownerName
+
+            globalUsage = globalUsage + chipUsage
+
+            -- For some reason, :IsPlayer() always returns false if obtained locally from entityMeta, it HAS to be called this way
+            if isValid( owner ) and owner:IsPlayer() then
+                ownerName = preparePlyName( getNick( owner ) )
+            else
+                owner = ID_WORLD
+                ownerName = ID_WORLD
+            end
+            
+            local id = rawget( idLookup, owner )
+
+            if not id then
+                idCount = idCount + 1
+                id = idCount
+                rawset( idLookup, owner, id )
+            end
+
+            local data = rawget( perPlyData, id )
+            local dataCount
+
+            if data then
+                dataCount = rawget( data, "Count" )
+                rawset( data, "OwnerUsage", rawget( data, "OwnerUsage" ) + chipUsage )
+            else
+                data = {
+                    Count = 0,
+                    Owner = owner,
+                    OwnerName = ownerName,
+                    OwnerUsage = chipUsage,
+                }
+
+                rawset( perPlyData, id, data )
+                dataCount = 0
+            end
+
+            dataCount = dataCount + 1
+            rawset( data, dataCount, chipName )
+            dataCount = dataCount + 1
+            rawset( data, dataCount, CHIP_SHORTHANDS[chipClass] )
+            dataCount = dataCount + 1
+            rawset( data, dataCount, chipUsage )
+
+            rawset( data, "Count", dataCount )
+        end
+    end
+
+    rawset( perPlyData, "Count", idCount )
+
+    net.Start( "CFC_ChipLister_UpdateListData" )
+    net.WriteTable( perPlyData ) -- I would use TableToJSON, Compress, and WriteData, but player and chip names could easily break the formatting
```

Comment:
TableToJSON should account for all of that, using the compress function allows you to cram much more into it. Currently having too many chips (might be so many that will never happen but still should be accounted for) will certainly exceed the message limit easily.